saleor/graphql/discount/types.py

```diff
@@ -21,28 +36,58 @@ class Sale(CountableDjangoObjectType):
     products = PrefetchingConnectionField(
         Product, description="List of products this sale applies to."
     )
-    translation = TranslationField(SaleTranslation, type_name="sale")
+    translation = TranslationField(
+        SaleTranslation,
+        type_name="sale",
+        resolver=ChannelContextType.resolve_translation,
+    )
+    channel_listing = graphene.List(
+        graphene.NonNull(SaleChannelListing),
+        description="List of channels available for the sale.",
+    )
+    discount_value = graphene.Float(description="Sale value.")
+    currency = graphene.String(description="Currency code for sale.")
 
     class Meta:
+        default_resolver = ChannelContextType.resolver_with_context
         description = (
             "Sales allow creating discounts for categories, collections or products "
             "and are visible to all the customers."
         )
         interfaces = [relay.Node]
         model = models.Sale
-        only_fields = ["end_date", "id", "name", "start_date", "type", "value"]
+        only_fields = ["end_date", "id", "name", "start_date", "type"]
 
     @staticmethod
-    def resolve_categories(root: models.Sale, *_args, **_kwargs):
-        return root.categories.all()
+    def resolve_categories(root: ChannelContext[models.Sale], *_args, **_kwargs):
+        return root.node.categories.all()
+
+    @staticmethod
+    @permission_required(DiscountPermissions.MANAGE_DISCOUNTS)
+    def resolve_channel_listing(root: ChannelContext[models.Sale], *_args, **_kwargs):
+        return root.node.channel_listing.all()
 
     @staticmethod
     def resolve_collections(root: models.Sale, info, **_kwargs):
```

Comment:
ChannelContext? 
If you don't find it we should add a test for query  `collections`  from Sale.